common/djangoapps/track/transformers.py

```diff
@@ -0,0 +1,484 @@
+            will be used, so if a transformer exists with a name of
+            `'edx.ui.lms.'` and another transformer has the name
+            `'edx.ui.lms.sequence.'` then an event called
+            `'edx.ui.lms.sequence.tab_selected'` will be handled by the
+            `'edx.ui.lms.sequence.'` transformer.
+
+            An *exact transformer* will only handle events whose name matches
+            name of the transformer exactly.
+
+            Exact transformers always take precedence over prefix transformers.
+
+            Transformers without a name will not be added to the registry, and
+            cannot
+            be accessed via the `EventTransformer.create_transformer()`
+            classmethod.
+
+        is_legacy_event:
+            If an event is a legacy event, it needs to set event_type to the
+            legacy name for the event, and may need to set certain event fields
+            to maintain backward compatiblity.  If an event needs to provide
+            legacy support in some contexts, `is_legacy_event` can be defined
+            as a property to add dynamic behavior.
+
+            Default: False
+
+        legacy_event_type:
+            If the event is or can be a legacy event, it should define
+            the legacy value for the event_type field here.
+
+    Processing methods.  Override these to provide the behavior needed for your
+    particular EventTransformer:
+
+        self.process_legacy_fields():
+            This method should modify the event payload in any way necessary to
+            support legacy event types.  It will only be run if
+            `is_legacy_event` returns a True value.
+
+        self.process_event()
+            This method modifies the event payload unconditionally.  It will
+            always be run.
+    """
+
+    # Properties to be overridden
+
+    is_legacy_event = False
+
+    @property
+    def legacy_event_type(self):
+        """
+        Override this as an attribute or property to provide the value for
+        the event's `event_type`, if it does not match the event's `name`.
+        """
+        raise NotImplementedError
+
+    # Convenience properties
+
+    @property
+    def name(self):
+        """
+        Returns the event's name.
+        """
+        return self[u'name']
+
+    @property
+    def context(self):
+        """
+        Returns the event's context dict.
+        """
+        return self[u'context']
+
+    @property
+    def event(self):
+        """
+        Return the primary payload of the event.
+        """
+        return self[u'event']
+
+    # Transform methods
+
+    def transform(self):
+        """
+        Transform the event with legacy fields and other necessary
+        modifications.
+        """
+        if self.is_legacy_event:
+            self.set_legacy_event_type()
+            self.process_legacy_fields()
+        self.process_event()
+
+    def set_legacy_event_type(self):
```

Comment:
Yeah.  This one's unambiguous, I think.  It's only used internally, and there's no particular reason to override it that I can think of.